### PR TITLE
Allow `gardenlet` to create its own `Gardenlet` resource for self-upgrades

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area ops-productivity
/kind bug

**What this PR does / why we need it**:
This PR extends the seed authorizer to allow `gardenlet` to create its own `seedmanagement.gardener.cloud/v1alpha1.Gardenlet` resource. Without this, it fails:

```
panic: failed creating seedmanagement.gardener.cloud/v1alpha1.Gardenlet object for self-upgrades: gardenlets.seedmanagement.gardener.cloud is forbidden: User "gardener.cloud:system:seed:some-seed" cannot create resource "gardenlets" in API group "seedmanagement.gardener.cloud" in the namespace "garden": only the following verbs are allowed for this resource type: [get list watch update patch]
```

**Which issue(s) this PR fixes**:
Missing in https://github.com/gardener/gardener/pull/9874/commits/1fe9707fd2f164b7c1128086e2c306f2b94d1cda which was part of https://github.com/gardener/gardener/pull/9874

**Special notes for your reviewer**:
/cc @ScheererJ @shafeeqes @timuthy 

**Release note**:

```bugfix operator
A bug has been fixed which prevented `gardenlet` from creating its own `seedmanagement.gardener.cloud/v1alpha1.Gardenlet` resource when `selfUpgrade` was set in its Helm chart values ([more information](https://github.com/gardener/gardener/blob/master/docs/deployment/deploy_gardenlet_manually.md#optional-enable-self-upgrades)).
```

### CODE REVIEW

File: pkg/admissioncontroller/webhook/admission/seedrestriction/handler.go [hunk 1]
```diff
@@ -71,24 +72,26 @@ func (h *Handler) Handle(ctx context.Context, request admission.Request) admissi
 
 	requestResource := schema.GroupResource{Group: request.Resource.Group, Resource: request.Resource.Resource}
 	switch requestResource {
-	case backupBucketResource:
-		return h.admitBackupBucket(ctx, seedName, request)
 	case backupEntryResource:
 		return h.admitBackupEntry(ctx, seedName, request)
+	case backupBucketResource:
+		return h.admitBackupBucket(ctx, seedName, request)
```
Comment (ScheererJ) [inline pkg/admissioncontroller/webhook/admission/seedrestriction/handler.go]: Is this automatically generated code or why was this section moved? With regards to alphabetic ordering it was better before.
WDYT?

**Review by ScheererJ — COMMENTED** [review]
Thanks for fixing the issue. I just have a minor question. Apart from that it already looks good.

/lgtm

File: pkg/admissioncontroller/webhook/admission/seedrestriction/handler.go (hunk 1 above)
Comment (rfranzke) [reply pkg/admissioncontroller/webhook/admission/seedrestriction/handler.go]: Sorry, reverted 😄

**Review by ScheererJ — APPROVED** [review]
/lgtm
/approve

**Pull request merged**